### ISSUE TITLE
Liquid Exception: undefined method `sort' for nil:NilClass

### ISSUE THREAD

**Issue by luixxiul** [jekyll/jekyll#3491]
By running ``jekyll serve`` with ``--limit_posts``, the code below occurs the undefined method error;

````
{% assign something = site.categories.example | sort: "title" %}
{% for post in something %}
    <li><a href="{{ post.url }}">{{ post.title }}</a></li>
{% endfor %}  
````
Is there any way to resolve it without changing the number of ``--limit_posts``?

**Comment by parkr** [comment]
Can you please run with `jekyll build --trace` and paste the output?

**Comment by luixxiul** [comment]
Okay, here is the output:
````
C:\Users\Admin\Jekyll> bundle exec jekyll serve --limit_posts 1 --trace
DL is deprecated, please use Fiddle
  Liquid Exception: undefined method `sort' for nil:NilClass in map/index.html
C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/filters.rb:248:in `sort': undefined method `sort' for nil:NilClass (NoMethodError)
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/strainer.rb:43:in `invoke'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/context.rb:82:in `invoke'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/variable.rb:50:in `block in render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/variable.rb:38:in `each'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/variable.rb:38:in `inject'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/variable.rb:38:in `render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/tags/assign.rb:26:in `render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/block.rb:109:in `block in render_all'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/block.rb:96:in `each'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/block.rb:96:in `render_all'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/block.rb:82:in `render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/template.rb:128:in `render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/liquid-2.6.2/lib/liquid/template.rb:138:in `render!'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/convertible.rb:106:in `render_liquid'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/convertible.rb:233:in `do_layout'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/page.rb:122:in `render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:298:in `block in render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:297:in `each'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:297:in `render'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:51:in `process'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/command.rb:28:in `process_site'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/commands/build.rb:56:in `build'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/commands/build.rb:34:in `process'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/lib/jekyll/commands/serve.rb:26:in `block (2 levels) in init_with_program'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/gems/jekyll-2.5.3/bin/jekyll:20:in `<top (required)>'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/bin/jekyll:23:in `load'
        from C:/Users/Admin/Jekyll/vendor/bundler/ruby/2.1.0/bin/jekyll:23:in `<main>'
````
The error itself can be avoided by adding ``{% if site.categories.example %}`` tag, though.

**Comment by MartinRogalla** [comment]
```diff
diff --git a/lib/jekyll/filters.rb b/lib/jekyll/filters.rb
index d9c84fe..7ef66ad 100644
--- a/lib/jekyll/filters.rb
+++ b/lib/jekyll/filters.rb
@@ -222,6 +222,9 @@ module Jekyll
     #
     # Returns the filtered array of objects
     def sort(input, property = nil, nils = "first")
+      if input.nil?
+        return input
+      end
       if property.nil?
         input.sort
       else
```
Avoids the issue, but I don't know if the sort function should be allowed to handle a null object.

**Comment by parkr** [comment]
Instead of returning `input`, maybe we raise a better error.

Thanks for finding that, @MartinRogalla.

**Comment by alfredxing** [comment]
An error would probably be better since if you're sorting something, chances are you'll be using it in a `for ` loop later on. I don't image that would play nicely with a `nil` value.

**Comment by MartinRogalla** [comment]
Fixed in #3520.
